### DISCUSSION TITLE
Decouple "explore on Superset" permission

### DISCUSSION THREAD

**Discussion by sfirke** [apache/superset#21074] [Ideas]
Noted at https://github.com/apache/superset/discussions/18807#discussioncomment-3368934, I and others would like to disable the public from being able to View Query and View Table on a chart in a dashboard.  The way I found to do that is to remove the `explore on Superset` permission from the Public role.

Unfortunately, that causes the Public user to no longer be able to view embedded standalone charts (not Dashboards) in iFrames.  I'm using that functionality on a webpage.

The request: decouple the permissions of being able to view a chart, view its query, and view it as a table.  Note that this affects other use cases besides the public user, see thread below.

**Comment by waleedshabbir1** [comment]
@sfirke for making charts public you also had to make dataset public?

**Reply by sfirke** [reply]
Yes, I believe so - I see I have given the Public role the permission of `datasource access on [mydb].mytable](id:11)]`

**Comment by eschutho** [comment]
cc @dpgaspar and @yousoph

**Comment by JohnDietrich-Pepper** [comment]
We aren't public but have a similar issue.  We want to enable View Table but NOT View Query as we view this as a security risk.  Explore also opens up other undesirable functionality.  Funny that this limitation exists when there are literally like 280 other permissions!
